mllib/src/main/scala/org/apache/spark/mllib/stat/test/AndersonDarlingTest.scala

```diff
@@ -0,0 +1,289 @@
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.2 / math.sqrt(n)) }
+    }
+  }
+
+  // Logistic distribution
+  class AndersonDarlingLogistic(val params: Seq[Double]) extends AndersonDarlingTheoreticalDist {
+    private val theoretical = new LogisticDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.25 -> 0.426, 0.10 -> 0.563, 0.05 -> 0.660,
+      0.025 -> 0.769, 0.01 -> 0.906, 0.005 -> 1.010
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.25 / n) }
+    }
+  }
+
+  // Weibull distribution
+  class AndersonDarlingWeibull(val params: Seq[Double]) extends AndersonDarlingTheoreticalDist {
+    private val theoretical = new WeibullDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.25 -> 0.474, 0.10 -> 0.637,
+      0.05 -> 0.757, 0.025 -> 0.877, 0.01 -> 1.038
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.2 / math.sqrt(n)) }
+    }
+  }
+
+  /**
+   * Perform a one sample Anderson-Darling test
+   * @param data data to test for a given distribution
+   * @param distName name of theoretical distribution: currently supports normal,
+   *            exponential, gumbel, logistic, weibull as
+   *            ['norm', 'exp', 'gumbel', 'logistic', 'weibull']
+   * @param params variable-length argument providing parameters for given distribution. When none
+   *               are provided, default parameters appropriate to each distribution are chosen. In
+   *               either case, critical values reflect adjustments that assume the parameters were
+   *               estimated from the data
+   * @return
+   */
+  @varargs
+  def testOneSample(data: RDD[Double], distName: String, params: Double*)
+    : AndersonDarlingTestResult = {
+    val n = data.count()
+    val dist = initDist(distName, params)
+    val localData = data.sortBy(x => x).mapPartitions(calcPartAD(_, dist, n)).collect()
+    val s = localData.foldLeft((0.0, 0.0)) { case ((prevStat, prevCt), (rawStat, adj, ct)) =>
+      val adjVal = 2 * prevCt * adj
+      val adjustedStat = rawStat + adjVal
+      val cumCt = prevCt + ct
+      (prevStat + adjustedStat, cumCt)
+    }._1
+    val ADStat = -1 * n - s / n
+    val criticalVals = dist.getCVs(n)
+    new AndersonDarlingTestResult(ADStat, criticalVals, NullHypothesis.OneSample.toString)
+  }
+
+
+  /**
+   * Calculate a partition's contribution to the Anderson-Darling statistic.
+   * In each partition we calculate 2 values, an unadjusted value that is contributed to the AD
+   * statistic directly, a value that must be adjusted by the number of values in the prior
```

Comment:
Yes, sorry, that value is adjust by a running aggregate of counts in prior partitions, so it should indeed read *partitions* not *partition*, corrected now. Thanks